test/Dapr.Actors.Test/ApiTokenTests.cs

```diff
@@ -16,107 +14,96 @@ namespace Dapr.Actors.Test
 {
     public class ApiTokenTests
     {
-        [Fact(Skip = "Failing due to #573")]
-        public void CreateProxyWithRemoting_WithApiToken()
+        [Fact]
+        public async Task CreateProxyWithRemoting_WithApiToken()
         {
+            await using var client = TestClient.CreateForMessageHandler();
+
             var actorId = new ActorId("abc");
-            var handler = new TestHttpClientHandler();
             var options = new ActorProxyOptions
             {
                 DaprApiToken = "test_token",
             };
-            var factory = new ActorProxyFactory(options, handler);
-            var proxy = factory.CreateActorProxy<ITestActor>(actorId, "TestActor");
-            var task = proxy.SetCountAsync(1, new CancellationToken());
 
-            handler.Requests.TryDequeue(out var entry).Should().BeTrue();
-            var headerValues = entry.Request.Headers.GetValues("dapr-api-token");
+            var request = await client.CaptureHttpRequestAsync(async handler =>
+            {
+                var factory = new ActorProxyFactory(options, handler);
+                var proxy = factory.CreateActorProxy<ITestActor>(actorId, "TestActor");
+                await proxy.SetCountAsync(1, new CancellationToken());
+            });
+
+            request.Dismiss();
+
+            var headerValues = request.Request.Headers.GetValues("dapr-api-token");
             headerValues.Should().Contain("test_token");
         }
 
-        [Fact(Skip = "Failing due to #573")]
-        public void CreateProxyWithRemoting_WithNoApiToken()
+        [Fact]
+        public async Task CreateProxyWithRemoting_WithNoApiToken()
         {
+            await using var client = TestClient.CreateForMessageHandler();
+
             var actorId = new ActorId("abc");
-            var handler = new TestHttpClientHandler();
-            var factory = new ActorProxyFactory(null, handler);
-            var proxy = factory.CreateActorProxy<ITestActor>(actorId, "TestActor");
-            var task = proxy.SetCountAsync(1, new CancellationToken());
-
-            handler.Requests.TryDequeue(out var entry).Should().BeTrue();
-            Action action = () => entry.Request.Headers.GetValues("dapr-api-token");
-            action.Should().Throw<InvalidOperationException>();
+
+            var request = await client.CaptureHttpRequestAsync(async handler =>
+            {
+                var factory = new ActorProxyFactory(null, handler);
+                var proxy = factory.CreateActorProxy<ITestActor>(actorId, "TestActor");
+                await proxy.SetCountAsync(1, new CancellationToken());
+            });
```

Comment:
This design approach lets us capture HTTP or gRPC traffic without timing issues. This is very robust and applies timeouts to everything that happens, so we can't end up with tests that hang indefinitely.